ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -422,6 +426,31 @@ renameSelection pwl WalletFace{..} walletSelRef name = do
 
   walletRefreshState
 
+-- | This function removes unknown keys from key file on startup, if user confirms that.
+checkUnknownKeys
+  :: PassiveWalletLayer IO
+  -> (ConfirmationType -> IO Bool)  -- ^ Asks user permission to delete unknown keys
+  -> IO ()
+checkUnknownKeys pwl confirmation = do
+  unknownRootIds <- pwlGetUnknownKeys pwl
+  unless (null unknownRootIds) $ do
+    let unknownRootIdsText =
+          T.dropEnd 1 . unlines $ T.drop 10 . sformat build <$> unknownRootIds
```

Comment:
This `T.drop 10` part looks pretty bad. If you want formatting different from what `build` generates, you should use a different formatter instead of removing some symbols. Now it's very fragile. E. g. someone may rename `HdRootId` and `HdRootId: ` will be changed to the new name else. Or we can decide that `Buildable` instance should be changed for another reason. Anyway, the point is that you can assume that `Buildable` instance produces something reasonable, but you can't assume that if you strip first 10 characters it will still be producing something reasonable.

Reply:
In this particular case I would recommend doing `_fromDb . getHdRootId`.

Reply:
Hmm. What would you recommend in common? I see that my solution is not beautiful and convenient, but OTOH I don't want to change an instance, because I usually believe that it makes sense and someone made it for some reason.

Reply:
```suggestion
          T.dropEnd 1 . unlines $ sformat hashHexF . _fromDb . getHdRootId <$> unknownRootIds
```

Reply:
I understand. I mean the common case :-)

Reply:
In common case you can define your own formatter. If you have type `X`, don't like its `Buildable` and don't want to change it, you can define `prettierXF :: Format r (X -> r)`.